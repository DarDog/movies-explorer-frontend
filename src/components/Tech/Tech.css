.tech {
  padding: 100px 70px;
  background-color: #F5F5F5;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.tech__title {
  margin: 20px 0 26px;
  padding: 0;
  font-weight: 400;
  font-size: 50px;
  line-height: 58px;
  text-align: center;
  letter-spacing: -0.04em;
  color: black;
}

.tech__paragraph {
  padding: 0;
  margin: 0 0 100px;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  text-align: center;
  letter-spacing: -0.04em;
  color: black;
}

.tech__list {
  width: 100%;
  max-width: 690px;
  justify-content: space-between;
  align-items: center;
  gap: 10px;
}

.tech__item {
  width: 100%;
  min-height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #E8E8E8;
  border-radius: 10px;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  letter-spacing: -0.04em;
  color: black;
}

@media screen and (max-width: 768px){
  .tech {
    padding: 90px 50px;
  }

  .tech__title {
    margin-top: 10px;
  }

  .tech__paragraph {
    font-size: 12px;
    line-height: 18px;
    margin-bottom: 80px;
    max-width: 460px;
  }

  .tech__list {

  }

  .tech__item {

  }
}
