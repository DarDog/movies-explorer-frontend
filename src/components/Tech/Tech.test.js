import React from 'react';
import Tech from "./Tech";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';


describe('Компонент Tech', () => {
  it('Рендерится', () => {
    render(<Tech/>);

    expect(screen.getByRole('heading', {level: 2})).toBeInTheDocument();
    expect(screen.getByText('Технологии')).toBeInTheDocument();
    expect(screen.getByRole('heading', {level: 3})).toBeInTheDocument();
    expect(screen.getByText('7 технологий')).toBeInTheDocument();
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getAllByRole('listitem')).toHaveLength(7);
  })
  it('Соответствует своему snapshot', () => {
    const component = render(<Tech/>);

    expect(component).toMatchSnapshot();
  })
})
