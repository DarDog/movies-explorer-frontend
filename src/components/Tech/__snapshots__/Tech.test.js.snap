// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Tech Соответствует своему snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="tech"
      >
        <h2
          class="title"
        >
          Технологии
        </h2>
        <h3
          class="tech__title"
        >
          7 технологий
        </h3>
        <p
          class="tech__paragraph"
        >
          На курсе веб-разработки мы освоили технологии, которые применили в дипломном проекте
        </p>
        <ul
          class="list tech__list "
        >
          <li
            class="tech__item"
          >
            HTML
          </li>
          <li
            class="tech__item"
          >
            CSS
          </li>
          <li
            class="tech__item"
          >
            JS
          </li>
          <li
            class="tech__item"
          >
            React
          </li>
          <li
            class="tech__item"
          >
            Git
          </li>
          <li
            class="tech__item"
          >
            Express.js
          </li>
          <li
            class="tech__item"
          >
            MongoDB
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="tech"
    >
      <h2
        class="title"
      >
        Технологии
      </h2>
      <h3
        class="tech__title"
      >
        7 технологий
      </h3>
      <p
        class="tech__paragraph"
      >
        На курсе веб-разработки мы освоили технологии, которые применили в дипломном проекте
      </p>
      <ul
        class="list tech__list "
      >
        <li
          class="tech__item"
        >
          HTML
        </li>
        <li
          class="tech__item"
        >
          CSS
        </li>
        <li
          class="tech__item"
        >
          JS
        </li>
        <li
          class="tech__item"
        >
          React
        </li>
        <li
          class="tech__item"
        >
          Git
        </li>
        <li
          class="tech__item"
        >
          Express.js
        </li>
        <li
          class="tech__item"
        >
          MongoDB
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
